### PR TITLE
[SCB-1711] support kie

### PR DESCRIPTION
**Pull request by GuoYL123**
(cherry picked from commit 833af5050ca3e26e75b327b499b0f9534f55d8b6)

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean install -Pit` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by liubao68** [comment]
How about dependen on client/kie-client module? 

I think for kie-client, the reactive performance is not important, and kie-client is quite simple, we can use the module for both spring-cloud-huawei and java-chassis.

**Comment by liubao68** [comment]
Please rename all your base package to ```org.apache.servicecomb.config.kie``` to avoid naming conflicts with classes in config-center module. check config-nacos module for a reference.

File: dynamic-config/config-kie/src/main/java/org/apache/servicecomb/config/kie/model/ValueType.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ */
+
+package org.apache.servicecomb.config.kie.model;
+
+/**
+ * @Author GuoYl123
+ * @Date 2020/1/8
+ **/
+public enum ValueType {
+  yml,
```
Comment (liubao68) [inline dynamic-config/config-kie/src/main/java/org/apache/servicecomb/config/kie/model/ValueType.java]: It's better to name constant with UPPER case, and add a value of string with LOWER case.

File: dynamic-config/config-kie/src/test/java/org/apache/servicecomb/config/kie/client/TestKieUtil.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+import mockit.Expectations;
+import mockit.Mocked;
+import org.apache.servicecomb.config.kie.model.KVResponse;
+import org.junit.Test;
+
+public class TestKieUtil {
+
+  @Test
+  public void testGetConfigByLabel(final @Mocked KVResponse resp) {
```
Comment (liubao68) [inline dynamic-config/config-kie/src/test/java/org/apache/servicecomb/config/kie/client/TestKieUtil.java]: Delete useless code

File: dynamic-config/config-kie/src/main/java/org/apache/servicecomb/config/kie/client/KieClient.java [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+      throw new IllegalStateException(e);
+    }
+    EXECUTOR
+        .scheduleWithFixedDelay(new ConfigRefresh(serviceUri), firstRefreshInterval,
+            refreshInterval, TimeUnit.SECONDS);
+  }
+
+  private void deployConfigClient() throws InterruptedException {
+    VertxOptions vertxOptions = new VertxOptions();
+    //todo :  how to deal with it
```
Comment (liubao68) [inline dynamic-config/config-kie/src/main/java/org/apache/servicecomb/config/kie/client/KieClient.java]: This TODO does not tell what is going to be done.

File: dynamic-config/config-kie/src/main/java/org/apache/servicecomb/config/kie/archaius/sources/KieConfigurationSourceImpl.java [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+import org.apache.commons.configuration.Configuration;
+import org.apache.servicecomb.config.ConfigMapping;
+import org.apache.servicecomb.config.kie.client.KieClient;
+import org.apache.servicecomb.config.kie.client.KieConfig;
+import org.apache.servicecomb.config.spi.ConfigCenterConfigurationSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @Author GuoYl123
```
Comment (WillemJiang) [inline dynamic-config/config-kie/src/main/java/org/apache/servicecomb/config/kie/archaius/sources/KieConfigurationSourceImpl.java]: Please remove the Author information, as we can look it up from SCM.

Comment (GuoYL123) [reply dynamic-config/config-kie/src/main/java/org/apache/servicecomb/config/kie/archaius/sources/KieConfigurationSourceImpl.java]: done

File: dynamic-config/config-kie/src/main/java/org/apache/servicecomb/config/kie/client/KieClient.java (hunk 3 above)
Comment (GuoYL123) [reply dynamic-config/config-kie/src/main/java/org/apache/servicecomb/config/kie/client/KieClient.java]: done

File: dynamic-config/config-kie/src/test/java/org/apache/servicecomb/config/kie/client/TestKieUtil.java (hunk 2 above)
Comment (GuoYL123) [reply dynamic-config/config-kie/src/test/java/org/apache/servicecomb/config/kie/client/TestKieUtil.java]: done

File: dynamic-config/config-kie/src/main/java/org/apache/servicecomb/config/kie/model/ValueType.java (hunk 1 above)
Comment (GuoYL123) [reply dynamic-config/config-kie/src/main/java/org/apache/servicecomb/config/kie/model/ValueType.java]: done

**Comment by GuoYL123** [comment]
I defined version in ./dependencies/default/pom.xml. But CI still failed ,and tell me version is missing. It‘s strange.

**Comment by liubao68** [comment]
retrigger CI

**Comment by liubao68** [comment]
retrigger CI

**Pull request merged**